### PR TITLE
Make sure the sshclient gets invoked so session is active

### PR DESCRIPTION
**Pull request by rkhamis**
Needs more testing

### CODE REVIEW

File: lib/JumpScale/clients/ssh/SSHClient.py [hunk 1]
```diff
@@ -25,10 +25,15 @@ def reset(self):
     def get(self, addr, port=22, login="root", passwd=None, stdout=True, forward_agent=True, allow_agent=True, look_for_keys=True,
             timeout=5, key_filename=None, passphrase=None, die=True, usecache=True):
         key = "%s_%s_%s_%s" % (addr, port, login, j.data.hash.md5_string(str(passwd)))
+        if key in self.cache and usecache:
+            try:
+                if not self.cache[key].transport.is_active():
+                    usecache = False
```
Comment (grimpy) [inline lib/JumpScale/clients/ssh/SSHClient.py]: if its not active anymore shouldnt we just pop it out of the cache?

Comment (grimpy) [inline lib/JumpScale/clients/ssh/SSHClient.py]: nevermind previous comment just noticed that even when setting usecache to false you still store the new connectin in the cache object.

Comment (rkhamis) [inline lib/JumpScale/clients/ssh/SSHClient.py]: yes. yes, we can.

Comment (zaibon) [inline lib/JumpScale/clients/ssh/SSHClient.py]: Can we document the argument so people understands what the behaviour is

File: lib/JumpScale/tools/executor/ExecutorSSH.py [hunk 2]
```diff
@@ -73,7 +74,12 @@ def sshclient(self):
                 else:
                     raise j.exceptions.RuntimeError("Could not find key:%s"%path)
 
-                self._sshclient.ssh_authorize("root",pubkey)
+            self._sshclient = j.clients.ssh.get(self.addr, self.port, login=self.login, passwd=self.passwd,
+                                                allow_agent=self.allow_agent, look_for_keys=self.look_for_keys,
+                                                key_filename=path, passphrase=None,
+                                                timeout=self.timeout, usecache=False)  # TODO: add passphrase
```
Comment (despiegk) [inline lib/JumpScale/tools/executor/ExecutorSSH.py]: please check @todo in this code see on branch ays_capnp or master branch
we need to document pushkey pubkey behaviour
is all very confusing

this should have been separate method for push key

the pubkey why do we need this?

also the =None ="" ... behaviour is wrong DO NOT USE
if pubkey:
   ....

needs to be all specific

**Pull request closed without merging**